examples/src/main/java/com/mapbox/navigation/examples/core/ReplayHistoryActivity.kt

```diff
@@ -165,6 +181,61 @@ class ReplayHistoryActivity : AppCompatActivity() {
         )
     }
 
+    /** Rendering the set route event **/
+
+    private val options: MapboxRouteLineOptions by lazy {
+        MapboxRouteLineOptions.Builder(this)
+            .withRouteLineResources(
+                RouteLineResources.Builder()
+                    .routeLineColorResources(
+                        RouteLineColorResources.Builder().build()
+                    )
+                    .build()
+            )
+            .withRouteLineBelowLayerId("road-label")
+            .withVanishingRouteLineEnabled(true)
+            .build()
+    }
+
+    private val routeLineView by lazy {
+        MapboxRouteLineView(options)
+    }
+
+    private val routeLineApi: MapboxRouteLineApi by lazy {
+        MapboxRouteLineApi(options)
+    }
+
+    private val routeArrowApi: MapboxRouteArrowApi by lazy {
+        MapboxRouteArrowApi()
+    }
+
+    private val routeArrowView: MapboxRouteArrowView by lazy {
+        MapboxRouteArrowView(
+            RouteArrowOptions.Builder(this)
+                .withAboveLayerId(RouteLayerConstants.PRIMARY_ROUTE_TRAFFIC_LAYER_ID)
+                .build()
+        )
+    }
+
+    private val routesObserver: RoutesObserver = RoutesObserver { routes ->
+        val routeLines = routes.map { RouteLine(it, null) }
+        routeLineApi.setRoutes(
+            routeLines
+        ) { value ->
+            binding.mapView.getMapboxMap().getStyle()?.apply {
+                routeLineView.renderRouteDrawData(this, value)
+            }
+        }
+    }
+
+    private val routeProgressObserver = RouteProgressObserver { routeProgress ->
+        routeLineApi.updateWithRouteProgress(routeProgress)
```

Comment:
You don't need this if your NOT using the vanishing route line feature.

Reply:
I did want to have vanishing route line in this, so let's make sure it's wired correctly. Looks like this was a missing piece https://github.com/mapbox/mapbox-navigation-android/pull/4637#discussion_r672674233

Reply:
Yes please! I want to replay some routes and double check if I can reproduce some vanishing issues I run into 🚀